File: LeetCode/283. Move Zeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        
        int zeros=0;
   
        
        for (int i=0;i<nums.Length;i++){
            
            if(i<nums.Length-zeros){
                 nums[i]=nums[i+zeros];
            }
            
            if(i>nums.Length-zeros-1){
                nums[i]=0;
            }
            else if(nums[i]==0){
                zeros++;
                
               Console.WriteLine(zeros);
            }
            
            
            
       if(i<nums.Length-zeros){ if(nums[i]==0) {i=i-1; }}
            
        }
        
    }
}

File: Data Structures and Algorithms/LeetCode/283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        for i in range(len(nums)-1,-1,-1):
            if nums[i]==0:
                nums.pop(i)
                nums.append(0)
        return nums        

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        j=-1
        for i in range(len(nums)-1):
            j+=1
            if nums[i]==0:
                if nums[i+1]==0:
                    j-=1
                    continue
                nums[j]=nums[i+1]
                nums[i+1]=0
        return nums        